### PR TITLE
refactor: update promote text for plugins.

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-468

### CODE REVIEW

File: packages/scripts/src/locale-kit/ast.ts
Comment (Jack-Works) [inline packages/scripts/src/locale-kit/ast.ts]: Move out to its own PR?

File: packages/.eslintrc.json
Comment (septs) [inline packages/.eslintrc.json]: why?

Comment (septs) [reply packages/.eslintrc.json]: this is security rule

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx]: Wait... You should use 

```js
import { useI18N } from '../locales'
const t = useI18N()
t.name()
```

Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx]: use i18n hooks.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInHistoryList.tsx
Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInHistoryList.tsx]: use TypedTrans from redpacket i18n folder.

File: packages/scripts/src/locale-kit/ast.ts
Comment (Jack-Works) [reply packages/scripts/src/locale-kit/ast.ts]: never mind. you can keep it  in this pr.

File: packages/.eslintrc.json
Comment (UncleBill) [reply packages/.eslintrc.json]: A trade-off between security and convenience.
Converting emoji into unicode make them hard to read and modify. We should look for another solution. I think scanning for zero-width characters is enough.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx]: Not all `<Trans />` here are in React component directly. Using `<Trans />` directly is more convenient.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInHistoryList.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/RedPacketInHistoryList.tsx]: done.

**Review by UncleBill — COMMENTED** [review]
I think a draft PR is not supposed to be reviewed except you get a request.

File: packages/.eslintrc.json
Comment (Jack-Works) [reply packages/.eslintrc.json]: Are emojis very common? If not, you can use eslint-disable

File: packages/mask/src/components/CompositionDialog/useSubmit.ts
Comment (Jack-Works) [inline packages/mask/src/components/CompositionDialog/useSubmit.ts]:
```suggestion
    const { t } = useI18N()
```

File: packages/mask/src/plugins/Gitcoin/SNSAdaptor/index.tsx
Comment (Jack-Works) [inline packages/mask/src/plugins/Gitcoin/SNSAdaptor/index.tsx:41]: Those two fields accepts I18NField. Please use i18nField (`{ i18nKey: '__description', fallback: '...' }`)

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: This field support I18NField.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (Jack-Works) [inline packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:128]: This field support I18NField.

File: packages/mask/src/components/CompositionDialog/useSubmit.ts
Comment (Jack-Works) [inline packages/mask/src/components/CompositionDialog/useSubmit.ts]: Maybe move these two lines into a function?

File: packages/plugins/FileService/src/index.ts
Comment (Jack-Works) [inline packages/plugins/FileService/src/index.ts]: Only export the util you want to expose.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: ReactNode one looks cleaner, doesn't it?
```diff
-            description: <Trans ns={PluginId.Gitcoin} i18nKey="description" />,
+            description: { i18nKey: `${PluginId.Gitcoin}:description`, fallback: 'description' },
```

In addition, all `ApplicationEntries` for Definition of all plugins are using `<Trans />` instead of `I18NField`, I think it's better to keep them align.

Comment (Jack-Works) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: If you're not using them by `t.description` or `Translate.description`, please make sure they start with `__`, in this case, `__description`.

We'll have a locale kit in the future to remove unused fields based on the type analysis, starting from `__` is a escape path, "don't remove this field even it is not used in the type level"

File: packages/.eslintrc.json
Comment (Jack-Works) [reply packages/.eslintrc.json]: ping

Comment (UncleBill) [reply packages/.eslintrc.json]: Not only emojis, but also some other symbols like Ξ ₿.

File: packages/mask/src/components/CompositionDialog/useSubmit.ts
Comment (UncleBill) [reply packages/mask/src/components/CompositionDialog/useSubmit.ts]: Need to pass down `t`, that will be complicated.

File: packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx
Comment (septs) [inline packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx:123]:
```suggestion
    }, [asset?.value, token, account, amount, expirationDateTime, isAuction])
```

File: packages/.eslintrc.json
Comment (Jack-Works) [reply packages/.eslintrc.json]: cc @septs can we have an allow list for this rule? It will be convenient.

File: packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx:123]: `t` is used. `t('promote_collectible')`

Comment (Jack-Works) [reply packages/mask/src/plugins/Collectible/SNSAdaptor/MakeOfferDialog.tsx:123]: IIRC `t` will change if the language changes.

**Comment by UncleBill** [comment]
Don't merge until #6181 gets merge into develop branch.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (Jack-Works) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: ping

File: packages/.eslintrc.json
Comment (Jack-Works) [reply packages/.eslintrc.json]: ping @septs

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: Instead of `__` prefix, can we define some reserved keys, such as `name`, `description`, etc, which can't be removed even they are not used, as each plugin needs to reserve them.


Our i18n keys are all in `snake_case`, we can force the reserved keys in `UpperCamelCase`, and normal keys in `lowerCamelCase` or `snake_case`.


Update: locale-kit has been removed, we can revisit it when we want it back.

Comment (Jack-Works) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: yes we can make some name blessed, but we still need a mark if devs want to use their own reserved names. `__` has already baked into the plugin infra

Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: TBH, `__` looks weird. 😂

Comment (Jack-Works) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: Do you have a better suggestion? We can use another prefix/suffix.

Comment (UncleBill) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: My suggestion is  using `UpperCamelCase` for reserved keys. (Or just capitalize the first letter)

File: packages/mask/src/plugins/Gitcoin/SNSAdaptor/index.tsx
Comment (UncleBill) [reply packages/mask/src/plugins/Gitcoin/SNSAdaptor/index.tsx:41]: `description` for Gitcoin plugin is so long, it will mess up the code.

```json
    "description": "Display specific information of Gitcoin projects, donate to a project directly on social media.",
```

```diff
-           description: <Trans ns={PluginId.Gitcoin} i18nKey="description" />,
+           description: {
+               i18nKey: 'description',
+               fallback:
+                   'Display specific information of Gitcoin projects, donate to a project directly on social media.',
+           },
```

BTW, the `fallback` can easily outdate without being noticed. And managing text/translation in too many places looks like not following the DRY principle.

Comment (Jack-Works) [reply packages/mask/src/plugins/Gitcoin/SNSAdaptor/index.tsx:41]: Agree. Track this issue on #6357.

File: packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx
Comment (Jack-Works) [reply packages/mask/src/plugins/RedPacket/SNSAdaptor/index.tsx:97]: Track this issue on #6357.

File: packages/.eslintrc.json
Comment (Jack-Works) [reply packages/.eslintrc.json]: This rule supports allow list. https://dimensiondev.github.io/eslint-plugin/src/rules/unicode/specific-set

Please allow characters you need to use here.

Note: For the code review perspective, `₿` should not be allowed because it looks visually like `B`

Comment (Jack-Works) [reply packages/.eslintrc.json]: I have no other change requests on this PR. Once this one fixed, I will approve it

**Review by Jack-Works — APPROVED** [review]

**Review by guanbinrui — APPROVED** [review]

**Pull request merged**
